compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -347,6 +348,14 @@ class Typer extends Namer
     findRefRecur(NoType, BindingPrec.NothingBound, NoContext)
   }
 
+  def toNotNullTermRef(tree: Tree, pt: Type)(implicit ctx: Context): Tree = tree.tpe match
```

Comment:
Let's add a comment for this method:
e.g.
```scala
If  `tree`'s type is a `TermRef` identified by flow typing to be non-null, then
cast away `tree`s nullability. Otherwise, `tree` remains unchanged.
```